File: p520.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int len = word.length();
        if (word[0] >= 'a' and word[0] <= 'z') {
            bool all_l = true;
            for (int i = 1; i < len; i++) {
                if (word[i] >= 'a' and word[i] <= 'z') {
                    // good
                } else {
                    all_l = false;
                }
            }
            return all_l;
        } else {
            bool all_l = true;
            bool all_u = true;
            for (int i = 1; i < len; i++) {
                if (word[i] >= 'a' and word[i] <= 'z') {
                    all_u = false;
                } else {
                    all_l = false;
                }
            }
            return all_l | all_u;
        }
    }
};

File: python/2023_daily/p520.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        # Check if all uppercase
        if all([char.isupper() for char in word]):
            return True

        # Check if all lowercase
        if all([char.islower() for char in word]):
            return True

        # Check if first char is uppercase and the rest chars are lowercase
        if word[0].isupper() and all([char.islower() for char in word[1:]]):
            return True

        return False
